sympy/core/mul.py

```diff
@@ -1258,32 +1261,72 @@ def _eval_pos_neg(self, sign):
     def _eval_is_negative(self):
         if self.args[0] == -1:
             return (-self).is_positive  # remove -1
-        return self._eval_pos_neg(-1)
-
+        return self._eval_pos_neg(-1)     
+    
     def _eval_is_odd(self):
+ 
+  """ First we check to see if the expression as a whole is an integer because
```

Comment:
Okay, I'll change that to comments.  Another thing I changed was in response to an error Travis was giving me that said: `AttributeError: 'NegativeOne' object has no attribute 'fraction'`.  I believe I fixed that with:
```
if coeff == 1 or coeff == -1:  
    r = True  
    for t in symbols.arg[1]:  
        if t.is_even:  

```